File: 1/prob01b.py
def fuel(m):
    f = int(m / 3) - 2
    if f > 0:
        f += fuel(f)
    return max(f, 0)


with open("input.txt", 'r') as f:
    data = f.read()

s = 0
for l in data.split():
    s += fuel(int(l))

print(s)

File: 2019/1/prob01b.rs
use std::fs;

fn main() {
    let mut sum = 0;

    let data = fs::read_to_string("input.txt").expect("Unable to read file");
    let nums = data.split("\n");
    for num in nums {
        sum += fuel(num.parse::<i32>().unwrap());
    }

    println!("{}", sum);
}

fn fuel(m: i32) -> i32 {
    let s: i32 = (m / 3) - 2;
    if s > 0 {
        return s + fuel(s.clone());
    }
    return 0;
}